docs/developer_guidelines.md

```diff
@@ -124,68 +125,79 @@ Resources:
+<type>[optional scope]: <subject>
 
-#### Full
+[optional body]
 
-Whenever the change you made needs more context you should use the full message
-format.
+[optional footer(s)]
+```
 
-The full message format is best summarized with the below rules:
+General rules that must be followed:
 
-- Separate the subject from the body with a blank line.
-- Limit the subject line to 50 characters.
-- Capitalize the subject line.
+- Every commit message must have a type and a subject. Scope, body and footer
+  sections are optional.
+- Separate the subject from the body with a blank line. The subject line should
+  be around 50 characters, the maximum is 72.
 - Do not end the subject line with a period.
 - Use the imperative mood in the subject line.
 - Wrap the body at 72 characters.
+- Add a body when the change needs more context.
+- Use the footer to provide links to the relevant GitHub issues.
 
-Below is a visual representation of the full message format:
+#### Types
 
-```
-Summarize changes in around 50 characters or less
+The type field tells us what kind of change a particular commit is doing.
 
-More detailed explanatory text, can be only a sentence or two.
-Wrap it to about 72 characters or so, editors like Visual Studio
-Code and Vim do this automatically. Treat the first line as the
-subject of the commit and the rest of the text as the body. The
-blank line separating the summary from the body is critical
-(unless you omit the body completely); various tools like `log`,
-`shortlog` and `rebase` can get confused if you run the two
-together.
+Must be one of the following:
 
-Explanatory text can be split into several paragraphs.
+- `feat`: A new feature was added.
+- `fix`: Bug was fixed.
+- `docs`: Only documentation was added, improved or removed (applies also to
+  source code documentation).
+- `refactor`: A code change that neither fixes a bug nor adds a feature.
+- `style`: Change was made that doesn't affect the meaning of the code (white
+  space, formatting, missing semi-colons, etc.)
+- `test`: Only tests were added, improved or removed.
+- `infra`: Anything related to the CI or automation changed. For example: any
+  change to the GitHub Action files, `.pre-commit.config.yaml` or any other of
+  the configuration files.
+- `chore`: Catch-all type. Use it when none of the other types fit.
 
-You can also include bullet points:
-* A
-* B
-* C
+#### Scope
 
-Optionally, reference relevant GitHub issues at the bottom,
-like this:
+The scope field is optional. Use it to provide information on what part of the
+codebase/functionality changed. Possible examples are: `driver`, `lwm2m_ctrl`,
+`ui`, `lora`, etc.
 
-Resolves: #123
-See also: #456, #789
-```
+General rules:
+
+- There is no list of valid scope fields. **Use what makes sense.**
+- **Use consistent scopes inside the project**. Look through the git log when
+  making changes on a project that a different developer worked on. Avoid cases
+  such as `driver` and `drivers`.
```

Comment:
Above you mention driver as a valid scope, but here you say to avoid it?
I guess you wanted to say that you should avoid mixing different forms of the same word (singular / plural), so consider rewording this sentence.

Reply:
Correct, I wanted to say that we should avoid using similar words (synonyms, singular/plural) for the same thing. I will reword it.